src/handlers.rs

```diff
@@ -22,68 +23,152 @@ impl fmt::Display for HandlerError {
+        }
+        Err(e @ ConfigurationError::Http(_)) => {
+            return Err(HandlerError::Other(e.into()));
+        }
+    };
+
+    if let Event::Issue(event) = event {
+        handle_issue(ctx, event, &config).await?;
+    }
+
+    if let Some(body) = event.comment_body() {
+        handle_command(ctx, event, &config, body).await?;
+    }
+
+    if let Err(e) = notification::handle(ctx, event).await {
+        log::error!(
+            "failed to process event {:?} with notification handler: {:?}",
+            event,
+            e
+        );
+    }
+
+    if let Err(e) = rustc_commits::handle(ctx, event).await {
+        log::error!(
+            "failed to process event {:?} with rustc_commits handler: {:?}",
+            event,
+            e
+        );
+    }
 
-        pub async fn handle(ctx: &Context, event: &Event) -> Result<(), HandlerError> {
-            let config = config::get(&ctx.github, event.repo_name()).await;
+    Ok(())
+}
 
+macro_rules! issue_handlers {
+    ($($name:ident,)*) => {
+        async fn handle_issue(ctx: &Context, event: &IssuesEvent, config: &Arc<Config>) -> Result<(), HandlerError> {
             $(
-            if let Some(input) = Handler::parse_input(
-                &$handler, ctx, event, config.as_ref().ok().and_then(|c| c.$name.as_ref()),
+            if let Some(input) = $name::parse_input(
+                ctx, event, config.$name.as_ref(),
             ).map_err(HandlerError::Message)? {
-                let config = match &config {
-                    Ok(config) => config,
-                    Err(e @ ConfigurationError::Missing) => {
-                        return Err(HandlerError::Message(e.to_string()));
-                    }
-                    Err(e @ ConfigurationError::Toml(_)) => {
-                        return Err(HandlerError::Message(e.to_string()));
-                    }
-                    Err(e @ ConfigurationError::Http(_)) => {
-                        return Err(HandlerError::Other(e.clone().into()));
-                    }
-                };
                 if let Some(config) = &config.$name {
-                    Handler::handle_input(&$handler, ctx, config, event, input).await.map_err(HandlerError::Other)?;
+                    $name::handle_input(ctx, config, event, input).await.map_err(HandlerError::Other)?;
                 } else {
                     return Err(HandlerError::Message(format!(
                         "The feature `{}` is not enabled in this repository.\n\
-                         To enable it add its section in the `triagebot.toml` \
-                         in the root of the repository.",
+                        To enable it add its section in the `triagebot.toml` \
+                        in the root of the repository.",
                         stringify!($name)
                     )));
                 }
             })*
+            Ok(())
+        }
+    }
+}
+
+issue_handlers! {
```

Comment:
Can you add some docs on when to add here or into the other handler? I think it's basically whether you need to respond to events other than opened/edited but would be good to document that.

Reply:
Done, I hope its clear enought.